server.js

```diff
@@ -1,38 +1,287 @@
+ *     parameters:
+ *      - name: trackName
+ *        in: query
+ *        required: false
+ *        format: string
+ *      - name: artistName
+ *        in: query
+ *        required: false
+ *        format: string
+ *      - name: genre
+ *        in: query
+ *        required: false
+ *        format: string
+ *      - name: year
+ *        in: query
+ *        required: false
+ *        format: string
+ *     tags: 
+ *      - Tracks
+ *     responses:
+ *       200:
+ *         description: Successful response.
+ */
+app.get('/tracks/', async (req, res) => {
+  let tracks = await Track.find(req.query);
+  if (tracks) {
+    // Get tracks for your party (i.e. danceability based on the number you specify).
+    // Test with different values. For example, choose 70 if you're looking for more energetic dances.
+    // Example: /tracks/?danceability=60
+    if (req.query.danceability) {
+      const songsByDanceability = await Track.find().gt(
+        'danceability',
+        req.query.danceability,
+      );
+      tracks = songsByDanceability;
+    }
+    res.json(tracks);
+  } else {
+    res.status(404).json({ error: 'Not found' });
+  }
+});
+
+// Get track by ID
+
+/**
+ * @swagger
+ * /tracks/id/{id}:
+ *   get:
+ *     summary: Get tracks by id
+ *     parameters:
+ *      - name: id
+ *        in: path
+ *        required: true
+ *        description: Get an id from the "tracks" query above. Click on "Try it out" and then fill in the song id.
+ *        format: integer
+ *     tags:
+ *      - Single result
+ *     responses:
+ *       200:
+ *         description: Successful response.
+ */
+app.get('/tracks/id/:id', async (req, res) => {
+  try {
+    const trackById = await Track.findById(req.params.id);
+    if (trackById) {
+      res.json(trackById);
+    } else {
+      res.status(404).json({ error: 'Track not found' });
+    }
+  } catch (err) {
+    res.status(400).json({ error: 'Invalid Id' });
+  }
+});
+
+// Get all songs by Billie Eilish
+
+/**
+ * @swagger
+ * /tracks/artist/billie_eilish:
+ *   get:
+ *     summary: Get all tracks by Billie Eilish
+ *     tags:
+ *      - Artist Collection
+ *     responses:
+ *       200:
+ *         description: Successful response.
+ */
+app.get('/tracks/artist/billie_eilish', (req, res) => {
+  Track.find({ artistName: 'Billie Eilish' }, (error, data) => {
+    if (error) {
+      res.status(404).json({ error: 'Not found' });
+    } else {
+      res.send(data);
+    }
+  });
+});
+
+// Get tracks from the pop genre
+
+/**
+ * @swagger
+ * /tracks/genre/pop:
+ *   get:
+ *     summary: Get all tracks within the pop genre
+ *     tags:
+ *      - Genre Collection
+ *     responses:
+ *       200:
+ *         description: Successful response.
+ */
+app.get('/tracks/genre/pop', (req, res) => {
+  Track.find({ genre: 'pop' }, (error, data) => {
+    if (error) {
+      res.status(404).json({ error: 'Not found' });
```

Comment:
Your error message could be more precise since you know what the user was looking for.

Reply:
True! Important observation. 💯